src/AzureAuth.Test/CommandMainTest.cs

```diff
@@ -384,6 +389,142 @@ public void TestPromptHintPrefixWithoutOption()
+        {
+            string filenameFromEnv = "normal_file_name_from_env";
+            this.envMock.Setup(env => env.Get("AZUREAUTH_CACHE_FILE")).Returns(filenameFromEnv);
+
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.WrappedCacheFilePath.Should().Be(filenameFromEnv);
+        }
+
+        /// <summary>
+        /// The test to evaluate the cache file name when both enviroment variable and option exist.
+        /// Ideally use option first.
+        /// </summary>
+        [Test]
+        public void TestCacheFileOptionWithFilenameFromEnvAndOption()
+        {
+            string filenameFromEnv = "normal_file_name_from_env";
+            this.envMock.Setup(env => env.Get("AZUREAUTH_CACHE_FILE")).Returns(filenameFromEnv);
+
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.CacheFilePath = "normal_file_name_from_option";
+
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.WrappedCacheFilePath.Should().Be("normal_file_name_from_option");
+        }
+
+        /// <summary>
+        /// The test to evaluate a default cache file name.
+        /// </summary>
+        [Test]
+        public void TestCacheFileOptionWithNoParameter()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.WrappedCacheFilePath.Should().Be($"msal_{subject.Tenant}.cache");
+        }
+
+        /// <summary>
+        /// The test to evaluate a relative cache path.
+        /// </summary>
+        [Test]
+        public void TestCacheFileOptionWithRelativePath()
+        {
+            CommandMain subject = this.serviceProvider.GetService<CommandMain>();
+            subject.Resource = "f0e8d801-3a50-48fd-b2da-6476d6e832a2";
+            subject.Client = "e19f71ed-3b14-448d-9346-9eff9753646b";
+            subject.Tenant = "9f6227ee-3d14-473e-8bed-1281171ef8c9";
+
+            string path = "../test/relative.cache";
+            subject.CacheFilePath = path;
+            subject.EvaluateOptions().Should().BeTrue();
+            subject.WrappedCacheFilePath.Should().Be(path);
+        }

```

Comment:
I think this test should assert failure and we force the caller to give us an absolute path.